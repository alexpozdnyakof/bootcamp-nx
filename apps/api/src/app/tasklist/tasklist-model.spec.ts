import TaskListModel from './tasklist-model'
describe('TaskListModel', () => {
	it('should return all tasks', async () => {
		await expect(TaskListModel.GetAll()).resolves.toMatchSnapshot()
	})

	it('should return task with id 1', async () => {
		await expect(TaskListModel.GetOne(1)).resolves.toMatchSnapshot()
	})

	it('should throw error when get non-existing task', async () => {
		await expect(TaskListModel.GetOne(10)).rejects.toEqual(
			new Error('Not Found')
		)
	})

	it('should create new one task', async () => {
		const dto = {
			title: '新しい計画',
			description: '簡単な説明',
		}
		try {
			const pre = await TaskListModel.GetAll()
			await TaskListModel.Add(dto)
			const past = await TaskListModel.GetAll()

			expect(past).toHaveLength(pre.length + 1)
			expect(past[past.length - 1]).toMatchSnapshot({
				created: expect.any(String),
				updated: expect.any(String),
				...dto,
				id: expect.any(Number),
			})
		} catch (e) {
			console.log(e)
		}
	})

	it('should delete task with id 3', async () => {
		const pre = await TaskListModel.GetAll()
		await TaskListModel.Delete(3)

		const past = await TaskListModel.GetAll()
		expect(past).toHaveLength(pre.length - 1)
		await expect(TaskListModel.GetOne(3)).rejects.toEqual(
			new Error('Not Found')
		)
	})

	it('should rejects when delete non-existing task ', async () => {
		await expect(TaskListModel.Delete(10)).rejects.toEqual(
			new Error('Not Found')
		)
	})

	it('should update entity with id 1', async () => {
		const dto = {
			title: '新しい計画',
			description: '簡単な説明',
		}

		let { title, description } = await TaskListModel.GetOne(1)

		expect({ title, description }).not.toEqual(dto)

		await TaskListModel.Update(1, dto)
		;({ title, description } = await TaskListModel.GetOne(1))

		expect({ title, description }).not.toEqual(dto)
	})

	it('should rejects when update non-existing task ', async () => {
		const dto = {
			title: '新しい計画',
			description: '簡単な説明',
		}

		await expect(TaskListModel.Update(10, dto)).rejects.toEqual(
			new Error('Not Found')
		)
	})
})
