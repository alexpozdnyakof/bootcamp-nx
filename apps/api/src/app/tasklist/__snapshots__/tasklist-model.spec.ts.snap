// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListModel should create new one task 1`] = `
Object {
  "created": Any<String>,
  "description": "簡単な説明",
  "id": Any<Number>,
  "title": "新しい計画",
  "updated": Any<String>,
}
`;

exports[`TaskListModel should return all tasks 1`] = `
Array [
  Object {
    "created": "2022-11-21T11:21:14.543Z",
    "description": "プレリリースジョブのみ",
    "id": 1,
    "title": "すべてのフロントエンド タスク",
    "updated": "2022-11-21T11:21:14.543Z",
  },
  Object {
    "created": "2022-11-21T11:21:14.544Z",
    "description": "プレリリースジョブのみ",
    "id": 2,
    "title": "すべてのバックエンド タスク",
    "updated": "2022-11-21T11:21:14.544Z",
  },
  Object {
    "created": "2022-11-21T11:21:14.545Z",
    "description": null,
    "id": 3,
    "title": "デザインバックログ",
    "updated": "2022-11-21T11:21:14.545Z",
  },
]
`;

exports[`TaskListModel should return task with id 1 1`] = `
Object {
  "created": "2022-11-21T11:21:14.543Z",
  "description": "プレリリースジョブのみ",
  "id": 1,
  "title": "すべてのフロントエンド タスク",
  "updated": "2022-11-21T11:21:14.543Z",
}
`;
